api/mesh/v1alpha1/dataplane_helpers.go

```diff
@@ -234,6 +234,9 @@ func (d *Dataplane_Networking_Outbound) GetTagsIncludingLegacy() map[string]stri
 			ServiceTag: d.Service,
 		}
 	}
+	if _, ok := d.Tags[ServiceTag]; !ok {
```

Comment:
Why is this necessary? I don't think both can be set.

Reply:
If you see this : https://app.circleci.com/pipelines/github/kumahq/kuma/23239/workflows/1774fae4-4952-4848-9da8-5e99749fd4b1/jobs/443891/parallel-runs/0/steps/0-106

It says that the VirtualHost has no name.
It is because in the test file https://github.com/kumahq/kuma/blob/master/pkg/xds/generator/outbound_proxy_generator_test.go#L744-L746

The old fashion service naming is used

Maybe it deserves a separated PR but to my understanding there are two options.

* Or the test is changed to use kuma.io/service tag so pkg/xds/generator/testdata/outbound-proxy/09.envoy.golden.yaml is correct
* Or the  GetTagsIncludingLegacy() is changed to expose the right name of the service because right now pkg/xds/generator/testdata/outbound-proxy/09.envoy.golden.yaml is not conformed

Reply:
OK let's open a separate PR for this

Reply:
Let's do this in this one : https://github.com/kumahq/kuma/pull/7197